deepmd/pt/model/atomic_model/pairtab_atomic_model.py

```diff
@@ -65,11 +79,16 @@ def __init__(
                 tab_info,
                 tab_data,
             ) = self.tab.get()  # this returns -> Tuple[np.array, np.array]
+            nspline, ntypes = tab_info[-2:].astype(int)
             self.register_buffer("tab_info", torch.from_numpy(tab_info))
-            self.register_buffer("tab_data", torch.from_numpy(tab_data))
+            self.register_buffer(
+                "tab_data",
+                torch.from_numpy(tab_data).reshape(ntypes, ntypes, nspline, 4),
+            )
         else:
             self.register_buffer("tab_info", None)
             self.register_buffer("tab_data", None)
+        self.bias_atom_e = None
```

Comment:
it's probably better to create a placeholder with a proper shape `ntypes, 1`. However, in the case we don't have an valid input file, the ntype is passed in via deserialization after initialization. We might need a way to fetch that information during initialization.

Reply:
We may want to ask the user to pass the type_map to `__init__`